src/CodeGen_LLVM.cpp

```diff
@@ -2857,7 +2857,18 @@ void CodeGen_LLVM::visit(const Call *op) {
         internal_assert(op->args.size() == 2);
         Expr x = op->args[0];
         Expr y = op->args[1];
-        Expr e = Internal::halide_exp(Internal::halide_log(x) * y);
+        Halide::Expr abs_x_pow_y = Internal::halide_exp(Internal::halide_log(abs(x)) * y);
+        Halide::Expr nan_expr = Halide::Internal::Call::make(
+            x.type(), "nan_f32", {}, Halide::Internal::Call::PureExtern);
+        Halide::Expr zero_or_neg_x = select(y == 0.0f,
```

Comment:
select has a variadic form that would clean this up a little. Also we should probably use integer mod rather than fmod, and we shouldn't compute both abs(x)^y and x^y. I suggest the following change:
```
Expr iy = floor(y);
Expr e = select(x >= 0.0f, abs_x_pow_y,    // positive x
                y == 0.0f, 1.0f,           // x^0 == 1
                iy != y, nan_expr,         // negative x to a non-integer power
                iy % 2 == 0, abs_x_pow_y,  // negative x to an even power
                -abs_x_pow_y);             // negative x to an odd power
```

Is the explicit handling of ^0 necessary? It seems like the even-y case would handle it. Does it do the right thing when x is nan?

Reply:
Very cool! didn't not select have a variadic form.
Why always compute |x| ? isn't faster to just do exp(log(x) *y) for strictly positive x ?
We need y==0.0f as fast path for x^0 otherwise we will not do the right thing for 0^0.

Reply:
What does IEEE754 define NaN^0 and Inf^0 to be?

Reply:
select evaluates both sides, so we'll be computing every possible RHS, and then mux between them using the conditions (e.g. because x could be a vector with a mix of positive and negative terms). That's why it's better to just do |x|.

Reply:
@steven-johnson Both are 1.0f e.g: 
```
    printf("%f\t%f\n", pow(nan(""),0.0f), pow(INFINITY, 0.0f));
```
prints 1.000000	1.000000